### DISCUSSION TITLE
Email sent although delivery error/SOFT FAIL

### DISCUSSION THREAD

**Discussion by d-nnis** [postalserver/postal#3017] [Help with using Postal]
## Describe the bug

We've got emails that apparently get delivered although delivery attempt was logged with "Temporary SMTP delivery error when sending...". When finally that email gets accepted from destination server ("sent") it gets delivered a second time. There are even emails who get delivered five times in that manner (four times status "error", finally with status "sent").
(The emails were reported by the customer)

See for example:
![grafik](https://github.com/postalserver/postal/assets/1454499/698db406-c267-49da-b851-e46966eeed5d)

Email headers of corresponding messages (keep in mind time difference between UTC (smtp-server) and "real"-time in customer-domain (+2h):

First failed attempt ("SOFT FAIL")
```
Received: from mx01.mailnotifications.example.com ([XXX.XXX.191.45])
  by mx3.customer-domain.com with ESMTP/TLS/ECDHE-RSA-AES256-GCM-SHA384; 07 Jun 2024 16:10:54 +0200
```
Third succesfull attempt ("SENT")
```
Received: from mx01.mailnotifications.example.com ([XXX.XXX.191.45])
  by mx3.customer-domain.com with ESMTP/TLS/ECDHE-RSA-AES256-GCM-SHA384; 07 Jun 2024 16:26:04 +0200
```

Although the following log is not part of the original event, the scheme is as follows with a timeout of about 60s:

```
Jun 25 12:24:30 mx01 postal-worker[1672860]: 2024-06-25 12:24:30 +0000 INFO   Connected to XXX.XXX.214.53:25 (mx3.example.com) log_id=MTWXPLID component=worker thread=work1 original_queued_message=995894 queued_message=995894
Jun 25 12:24:30 mx01 postal-worker[1672860]: 2024-06-25 12:24:30 +0000 INFO   Sending message 1::995663 to FIRST.NAME@example.com log_id=MTWXPLID component=worker thread=work1 original_queued_message=995894 queued_message=995894
...
Jun 25 12:25:20 mx01 postal-worker[1672860]: 2024-06-25 12:25:20 +0000 ERROR  Net::ReadTimeout: Net::ReadTimeout with #<TCPSocket:fd 14, AF_INET, 10.98.1.24, 52570> log_id=MTWXPLID component=worker thread=work1 original_queued_message=995894 queued_message=995894
```

## To Reproduce

This behaviour seems to be happening with that specific customer, i.e. their mail-server-system and seems to occur repeatedly

## Expected behaviour

I would expect that emails commented with "delivery error" would not get delivered - taken aside whatever the issue is with the slow/ unsuccesfull attempt to send the email.
(Is there an option to extend the attempt to deliver an email/ stretch the timeout a bit?)

## Environment details 

 - OS: Ubuntu 20.04.6 LTS
 - Postal 3.3.3

## Additional information/context

Please let me know if I can provide additional information. Thank you!

**Comment by willpower232** [comment]
I presume you're able to email other people without any issue? Do you know much about this particular customers email server? What software or provider they're using?

**Comment by d-nnis** [comment]
They are using MS Outlook servers and it is a major company. And yes, sending mails to other people/servers works fine.

**Reply by jaydeepnp** [reply]
I updated postal.yml with below configuration and took restart of postal service.

smtp_client:
read_timeout: 120

This resolved my issue.

**Comment by d-nnis** [comment]
Upgraded to Postal 3.3.4
production environment (Rails 7.0.8.1), Puma version: 6.4.2 (ruby 3.2.2-p53)

I'll keep watching the logs to see if the issue persists.

**Comment by rakurtz** [comment]
I am encountering the exact same issue with emails sent to my own personal email address (@icloud.com). the email get's delivered, but there will be a status of `softfail` with this kind of error message:

```
Net::ReadTimeout with #<TCPSocket:fd 12, AF_INET, 162.55.xxx.xxx, 42290>
Time: 45.1s
Support Ref: GRK6BDOT
```

i get the E-Mail though. sometimes twice or even three times. 

this doesnt happen to all emails, but i get the feeling this happens to emails with attachments of size 5 MB +.

any idea?

**Reply by d-nnis** [reply]
The last known SOFT FAIL of an email-sending + still delivered was an email less than 5kB. So I cannot confirm that the size is the problem.

**Reply by arthurzenika** [reply]
We are getting similar error messages 

```
Net::ReadTimeout with #<TCPSocket:fd 17, AF_INET, x.x.x.x, 60984>
Time: 35.31s
Support Ref: RPYRHKR4
```
And https://mxtoolbox.com/ indicates that the connection time and transaction start time are too slow for the destination receiving server. 

Is there a way the timeout of the ruby code that connects could be configurable ?

**Reply by jaydeepnp** [reply]
We are also observing similar error message while delivering  mails to one specific domain all other mails works fine.

Net::ReadTimeout with #<TCPSocket:fd 12, AF_INET, 10.130.xx.xxx, 45494>
Time: 46.23s
Support Ref: 243NW0RX

**Reply by jaydeepnp** [reply]
I updated postal.yml with below configuration and took restart of postal service. 

smtp_client:
  read_timeout: 120

This resolved my issue.

**Comment by i-know-nothing-Help-Me** [comment]
postal is a crap.. they think non existed domain is soft fail, may be in future it will get email, and keep on retying forever, getting thousands of bounce.
